packages/db/tests/search/test-RuleSearch.js

```diff
@@ -0,0 +1,271 @@
+      provider_cumulus_id: t.context.providerCumulusId,
+    }))
+  ));
+
+  await t.context.rulePgModel.insert(
+    t.context.knex,
+    rules
+  );
+});
+
+test.after.always(async (t) => {
+  await destroyLocalTestDb({
+    ...t.context,
+    testDbName,
+  });
+});
+
+test('RuleSearch returns correct response for basic query', async (t) => {
+  const { knex } = t.context;
+  const dbSearch = new RuleSearch({});
+  const results = await dbSearch.query(knex);
+  t.is(results.meta.count, 50);
+  t.is(results.results.length, 10);
+
+  const expectedResponse1 = {
+    name: 'fakeRule-0',
+    createdAt: new Date(2017, 11, 31).getTime(),
+    updatedAt: new Date(2018, 0, 1).getTime(),
+    state: 'ENABLED',
+    rule: {
+      type: 'onetime',
+    },
+    workflow: 'testWorkflow-0',
+    collection: {
+      name: 'testCollection',
+      version: '8',
+    },
+    provider: t.context.testProvider.name,
+  };
+
+  const expectedResponse10 = {
+    name: 'fakeRule-9',
+    createdAt: new Date(2017, 11, 31).getTime(),
+    updatedAt: new Date(2018, 0, 1).getTime(),
+    state: 'DISABLED',
+    rule: {
+      type: 'onetime',
+    },
+    workflow: 'testWorkflow-9',
+    collection: {
+      name: 'testCollection',
+      version: '8',
+    },
+    provider: t.context.testProvider.name,
+  };
+
+  t.deepEqual(results.results[0], expectedResponse1);
+  t.deepEqual(results.results[9], expectedResponse10);
+});
+
+test('RuleSearchsupports page and limit params', async (t) => {
+  const { knex } = t.context;
+  let queryStringParameters = {
+    limit: 25,
+    page: 2,
+  };
+  let dbSearch = new RuleSearch({ queryStringParameters });
+  let response = await dbSearch.query(knex);
+  t.is(response.meta.count, 50);
+  t.is(response.results?.length, 25);
+
+  queryStringParameters = {
+    limit: 10,
+    page: 5,
+  };
+  dbSearch = new RuleSearch({ queryStringParameters });
+  response = await dbSearch.query(knex);
+  t.is(response.meta.count, 50);
+  t.is(response.results?.length, 10);
+
+  queryStringParameters = {
+    limit: 10,
+    page: 11,
+  };
+  dbSearch = new RuleSearch({ queryStringParameters });
+  response = await dbSearch.query(knex);
+  t.is(response.meta.count, 50);
+  t.is(response.results?.length, 0);
+});
+
+test('RuleSearch supports infix search', async (t) => {
+  const { knex } = t.context;
+  const queryStringParameters = {
+    limit: 50,
+    infix: 'Rule-27',
+  };
+  const dbSearch = new RuleSearch({ queryStringParameters });
+  const response = await dbSearch.query(knex);
+  t.is(response.meta.count, 1);
+  t.is(response.results?.length, 1);
+});
+
+test('RuleSearch supports prefix search', async (t) => {
+  const { knex } = t.context;
+  const queryStringParameters = {
+    limit: 50,
+    prefix: 'fakeRule-1',
+  };
+  const dbSearch = new RuleSearch({ queryStringParameters });
+  const response = await dbSearch.query(knex);
+  t.is(response.meta.count, 11);
+  t.is(response.results?.length, 11);
+});
+
```

Comment:
Fair question. I'll go through these again. My thinking here was along these lines:

If we're testing the RuleSearch class for e.g. does it support a term search for a boolean field? Is that really implemented in the RuleSearch class? Or is that actually implemented in the BaseClass and tested there? Same gist for range, number field, etc.

Then the others appear to be testing the field-mappings and ensuring that the correct fields are added. That's... maybe good? It's a test to ensure I added the fields there. I was originally thinking that wasn't an especially useful test but I do see your point.

I think the biggest argument for adding these is "do what the other records have done". I completely buy that and will re-evaluate 😄

Reply:
Oh you're right. I was assuming BaseSearch had tests and it doesn't. It does feel like a lot of these assertions belong there instead of the child classes but I get it now. BaseSearch is abstract and can't be instantiated directly so testing it would mean exporting it. 

I think in order to not rock the boat I'll take your suggestion and add tests to my child class. It's possible this could come up again though and might warrant a refactor at some point. That or there's yet more I'm missing 😄

Reply:
https://github.com/nasa/cumulus/pull/3760/commits/2c4654ba583933c2f93fafb35d1bd9a225bb4b76 Added the tests mentioned above with one exception. There aren't _really_ number fields on Rules. https://github.com/nasa/cumulus/pull/3760/commits/2c4654ba583933c2f93fafb35d1bd9a225bb4b76